### PR TITLE
Fixed warnings seen when compiling inside dEQP.

### PR DESCRIPTION
**Pull request by asuonpaa**
Had to do some warning fixes when updating Amber to latest version for dEQP.

### CODE REVIEW

File: kokoro/scripts/windows/build.bat [hunk 1]
```diff
@@ -18,8 +18,8 @@ set BUILD_ROOT=%cd%
 set SRC=%cd%\github\amber
 set BUILD_TYPE=%1
 
-choco install cmake --pre --yes --no-progress
-choco upgrade cmake --pre --yes --no-progress
+choco install cmake --pre --yes --no-progress --version 3.15.4
```
Comment (dj2) [inline kokoro/scripts/windows/build.bat]: Why is this needed? This has been working on the bots for the last few weeks?

File: src/verifier.cc [hunk 2]
```diff
@@ -25,29 +25,28 @@
...
     ++src;
-    src_bit_offset = static_cast<uint8_t>(src_bit_offset - kBitsPerByte);
+    src_bit_offset = src_bit_offset - kBitsPerByte;
   }
 
   // Number of bytes greater than or equal to |(src_bit_offset + bits) / 8|.
-  const uint8_t size_in_bytes =
-      static_cast<uint8_t>((src_bit_offset + bits + 7) / kBitsPerByte);
-  assert(size_in_bytes <= static_cast<uint8_t>(kBitsPerByte));
+  const uint32_t size_in_bytes = (src_bit_offset + bits + 7) / kBitsPerByte;
```
Comment (dj2) [inline src/verifier.cc]: These were all casted correctly, what was the error from CTS?

File: kokoro/scripts/windows/build.bat (hunk 1 above)
Comment (paulthomson) [reply kokoro/scripts/windows/build.bat]: [Here is one failure](https://source.cloud.google.com/results/invocations/ed781401-e336-43ef-8214-088a4def18ff/targets/graphics_shader_compiler%2Famber%2Fwindows-msvc-2017-debug%2Fpresubmit/log)

I am guessing a new pre-release version of CMake was made available that fails to download. But it might not be this. I suggested specifying the latest stable release. I also suggested removing the `--pre` flag.

Comment (dj2) [reply kokoro/scripts/windows/build.bat]: If we drop the --pre, can we drop the --version 3.15.4? That way we don't pin to a specific version we just get the latest released version?

File: src/verifier.cc (hunk 2 above)
Comment (asuonpaa) [reply src/verifier.cc]:
```
[ 34%] Building C object framework/qphelper/CMakeFiles/qphelper.dir/qpInfo.c.o
/home/ari/git/issueNewAmber/vk-gl-cts/external/amber/src/src/verifier.cc: In function ‘std::vector<double> amber::{anonymous}::GetActualValuesFromTexel(const uint8_t*, const amber::Format*)’:
/home/ari/git/issueNewAmber/vk-gl-cts/external/amber/src/src/verifier.cc:271:36: error: conversion from ‘uint32_t’ {aka ‘unsigned int’} to ‘uint8_t’ {aka ‘unsigned char’} may change value [-Werror=conversion]
       bit_offset += seg.GetNumBits();
                                    ^
/home/ari/git/issueNewAmber/vk-gl-cts/external/amber/src/src/verifier.cc:323:16: error: conversion from ‘uint32_t’ {aka ‘unsigned int’} to ‘uint8_t’ {aka ‘unsigned char’} may change value [-Werror=conversion]
     bit_offset += num_bits;
     ~~~~~~~~~~~^~~~~~~~~~~
```

File: kokoro/scripts/windows/build.bat (hunk 1 above)
Comment (asuonpaa) [reply kokoro/scripts/windows/build.bat]: Let's try that. Can someone add the tag?
